### PR TITLE
[MM-22622] Add mentionHighlightDisabled props on create post and create comment

### PR DESCRIPTION
**Pull request by fm2munsh**
#### Summary
- Adds a check in the markdown component to set `options.mentionHighlight` to `false` if  `props.mentionHighlightDisabled` is `true`
- Sets `mentionHighlightDisabled` to `true` when creating a comment or post 
(this is also checked on the server but it is also needed on the webapp or else the `@ channel` text will temporarily flash as highlighted before the message gets validated on the server) 
- Tests

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22622

#### Related Pull Requests
- Has Server changes: https://github.com/mattermost/mattermost-server/pull/13902

#### Screenshots
Screen Shot 2020-02-15 at 3 12 47 PM

### CODE REVIEW

File: components/create_comment/create_comment.jsx [hunk 1]
```diff
@@ -430,12 +430,25 @@ class CreateComment extends React.PureComponent {
...
-        if (notificationsToChannel &&
-            membersCount > Constants.NOTIFY_ALL_MEMBERS &&
+        const {channelMembersCount, enableConfirmNotificationsToChannel, useChannelMentions, isTimezoneEnabled} = this.props;
+        const {draft} = this.state;
+        if (!useChannelMentions && containsAtChannel(draft.message, {checkAllMentions: true})) {
+            const updatedDraft = {
+                ...draft,
+                props: {
+                    ...draft.props,
+                    mentionHighlightDisabled: true,
```
Comment (hmhealey) [inline components/create_comment/create_comment.jsx]: Can we trust the client to set this correctly? What if a user that can't do channel-wide mentions were to use the API to manually send a post without this field? It feels to me like this is something that the server would set.

Comment (fm2munsh) [reply components/create_comment/create_comment.jsx]: The server pr (mattermost/mattermost-server#13902) also ensures the props are set if the user is lacking permission to channel mention.

The reason also have it here is for the split second immediately after creating the post but before the post is actually validated on the server. During that time the message would show up as highlighted (for the user sending the post) if the client did not also set the prop.

**Comment by fm2munsh** [comment]
@srkgupta I dont think the first issue you highlighted is related to my PR since I get that same console error on community.mattermost.com and my local master branch 
Screen Shot 2020-03-02 at 12 11 55 PM

I think a new ticket for the console error may be more appropriate.

As for the other two issues (channel header and editing messages with mentions) I brought them up with @michaelgamble in a UX review and based on what he thinks the appropriate UX for it will be I will modify the behaviour for those two cases.

**Comment by fm2munsh** [comment]
@srkgupta After looking into it some more that console error seems to be an error coming from the Lastpass chrome extension - if you disable lastpass on the page and then try sending a message you wont see the error

**Comment by srkgupta** [comment]
Thanks @fm2munsh. Yes indeed it was due to the LastPass Extension. The issue was actually not seen on my incognito window where the LastPass extension wasnt enabled. I disabled it on my main window as well and the error was no longer seen. Thanks for checking it.

**Comment by fm2munsh** [comment]
@srkgupta I have resolved the second issue - The behaviour implemented is that the `channel header` will ignore `@ all` `@ here` `@ channel` in all cases and treat them as normal text regardless of user permission since it was never triggering notifications anyway. 

As for the third issue of editing a post I discussed it with @michaelgamble and since currently notifications do _not_ get triggered when editing a post we will open a new ticket to handle sending notifications when adding mentions to edited posts. (I.e. It will not be handled as part of this ticket and the behaviour we have right now is sufficient)

**Comment by srkgupta** [comment]
Thanks @fm2munsh I have raised that issue separately here:
https://mattermost.atlassian.net/browse/MM-22866

**Pull request merged**